### ISSUE TITLE
slower result when using omp 

### ISSUE THREAD

**Issue by zzycarrot** [ssloy/tinyrenderer#140]
when writting the Triangle() function ,i get a much slower result when i used omp parallel,(2 times slower infact),
<img width="800" alt="fd653ba9ace43bf3f707a4f73c0e103" src="https://github.com/ssloy/tinyrenderer/assets/87789666/5efb9184-b29d-45b5-872c-f3a9907037cb">

**Comment by DouYuexi** [comment]
我已经收到您的邮件了，我将会尽快处理，保持联系！

**Comment by zzycarrot** [comment]
is it because the variable "image" was locked when it was been written that other threads can't use it ,which resulted in a slower speed than serial computing?

**Comment by ssloy** [comment]
Is the behavior same with a simple #pragma omp parallel for ?
(i mean without collapse/scheduling)

**Comment by zzycarrot** [comment]
> Is the behavior same with a simple #pragma omp parallel for ? (i mean without collapse/scheduling)

yes it is, i've tried a simple #pragma omp parallel for but with the same result

**Comment by ssloy** [comment]
I am curious. Can you try the code from master branch in my repository with and without this #pragma ?
https://github.com/ssloy/tinyrenderer/blob/6d3acdb9ee452449b528ef3900ef5528dad0f807/our_gl.cpp#L41

**Comment by zzycarrot** [comment]
> I am curious. Can you try the code from master branch in my repository with and without this #pragma ?
> 
> https://github.com/ssloy/tinyrenderer/blob/6d3acdb9ee452449b528ef3900ef5528dad0f807/our_gl.cpp#L41
i've tried yours now and it's performing as it should be (faster with omp) , maybe i've written my code wrongly ,

**Comment by ssloy** [comment]
i see nothing wrong with your code, if you find the problem please share
